src/lib.rs

```diff
@@ -73,8 +73,8 @@ impl<'a, 'b> plugin::Plugin<Request<'a, 'b>> for Raw {
 
         if need_read {
             let max_length = req.get::<persistent::Read<MaxBodyLength>>()
-                .ok().cloned().unwrap_or(DEFAULT_BODY_LIMIT);
-            let body = try!(read_body_as_utf8(req, max_length));
+                .ok().unwrap_or(std::sync::Arc::new(DEFAULT_BODY_LIMIT));
```

Comment:
Compiling bodyparser v0.0.4 (file:///Users/sfusco/code/lab/rustweb/reklis-body-parser)
src/lib.rs:76:23: 76:31 error: type `core::option::Option<alloc::arc::Arc<usize>>` does not implement any method in scope named `cloned`
src/lib.rs:76                 .ok().cloned().unwrap_or(DEFAULT_BODY_LIMIT);
                                    ^~~~~~~~
error: aborting due to previous error
Could not compile `bodyparser`.